applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,442 @@
+            array_1d<double, 3> reaction =
+                rGeometry[0].FastGetSolutionStepValue(rReactionVariable) /
+                rGeometry[0].GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS);
+            for (unsigned int i_node = 1; i_node < rGeometry.PointsNumber(); ++i_node)
+            {
+                const NodeType& r_node = rGeometry[i_node];
+                noalias(reaction) += r_node.FastGetSolutionStepValue(rReactionVariable) /
+                                     r_node.GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS);
+            }
+
+            return CalculateReactionBasedYPlusUTau(
+                rFrictionVelocity, reaction, rNormal, Density, KinematicViscosity, WallHeight);
+        };
+
+    CalculateYPlusAndUTauForConditions(rModelPart, rKinematicViscosityVariable, method);
+
+    KRATOS_CATCH("");
+}
+
+void CalculateYPlusAndUTauForConditionsBasedOnLinearLogarithmicWallFunction(
+    ModelPart& rModelPart,
+    const Variable<double>& rKinematicViscosityVariable,
+    const double VonKarman,
+    const double WallSmoothness,
+    const int MaxIterations,
+    const double Tolerance)
+{
+    KRATOS_TRY
+
+    const std::function<double(array_1d<double, 3>&, const GeometryType&, const array_1d<double, 3>&,
+                               const Vector&, const double, const double, const double)>
+        method = [VonKarman, WallSmoothness, MaxIterations, Tolerance](
+                     array_1d<double, 3>& rFrictionVelocity,
+                     const GeometryType& rGeometry, const array_1d<double, 3>& rNormal,
+                     const Vector& rGaussShapeFunctions, const double Density,
+                     const double KinematicViscosity, const double WallHeight) {
+            const array_1d<double, 3>& r_wall_velocity =
+                EvaluateInPoint(rGeometry, VELOCITY, rGaussShapeFunctions);
+            const array_1d<double, 3>& r_mesh_velocity =
+                EvaluateInPoint(rGeometry, MESH_VELOCITY, rGaussShapeFunctions);
+
+            return CalculateLinearLogarithmicWallFunctionBasedYPlusAndUtau(
+                rFrictionVelocity, r_wall_velocity - r_mesh_velocity, rNormal, KinematicViscosity,
+                WallHeight, VonKarman, WallSmoothness, MaxIterations, Tolerance);
+        };
+
+    CalculateYPlusAndUTauForConditions(rModelPart, rKinematicViscosityVariable, method);
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void DistributeConditionVariableToNodes(ModelPart& rModelPart,
+                                        const Variable<TDataType>& rVariable)
+{
+    KRATOS_TRY
+
+    CalculateNumberOfNeighbourConditions(rModelPart);
+
+    ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetNonHistoricalVariableToZero(rVariable, r_nodes);
+
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
```

Comment:
I would advise using the loop as in the parallel utilities:

https://github.com/KratosMultiphysics/Kratos/wiki/Parallel-Utilities

that would make your code cleaner

Reply:
done :) It is really clean now :)